command/git.go

```diff
@@ -277,7 +280,27 @@ func (c *gitFetchProject) buildModuleCloneCommand(conf *model.TaskConfig, opts c
 }
 
 // Execute gets the source code required by the project
-func (c *gitFetchProject) Execute(ctx context.Context,
+// Retries some number of times before failing
+func (c *gitFetchProject) Execute(ctx context.Context, comm client.Communicator, logger client.LoggerProducer, conf *model.TaskConfig) error {
+	var err error
+	for iter := 1; iter <= GitFetchProjectRetries; iter++ {
+		err = c.executeLoop(ctx, comm, logger, conf)
+		if err == nil {
+			return nil
+		}
+		grip.Error(message.WrapError(err, message.Fields{
```

Comment:
Log this to the task's logger. Though I think you only need to log this when you use up all the retries, or reword it to say that it will retry